inductiva/coastal/bathymetry.py

```diff
@@ -355,30 +370,40 @@ def plot(
             path: Path to save the plot. If `None`, the plot is not saved, and
               the matplotlib `Axes` object is returned instead.
             x_resolution: Resolution, in meters, of the plotting grid in the x
-              direction.
+              direction. If `None`, the resolution is determined by the
+              bathymetry
             y_resolution: Resolution, in meters, of the plotting grid in the y
-              direction.
+              direction. If `None`, the resolution is determined by the
+              bathymetry
             threshold_distance: Threshold distance to filter out points on the
               uniform grid that are far from points where the bathymetry is
               defined.
         """
 
-        if not self.is_uniform_grid():
-            logging.info("Plotting the bathymetry on a uniform grid...")
+        if self.is_uniform_grid(
+        ) and x_resolution is None and y_resolution is None:
+            x_size = self.x_num()
+            y_size = self.y_num()
 
-        depths_grid, _ = self._interpolate_to_uniform_grid(
-            x_resolution,
-            y_resolution,
-            threshold_distance=threshold_distance,
-            nullable=True,
-        )
+            depths_grid = self.depths.reshape((x_size, y_size))
```

Comment:
Why you need this reshape? Isn't the bathymetry already like this?

Reply:
No, the bathymetry is a one dimensional array

Reply:
1D array. That was unexpected...
OK.